### PR TITLE
feat(specs-list): improving focus management and keyboard handling of the specs list.

### PR DESCRIPTION
**Pull request by JessicaSachs**
I'm going to switch onto the Specs List on the Index page in a moment, so I want to get this in here and then do the refactor to make it generic in another PR

https://user-images.githubusercontent.com/2801156/144541181-5251fb1e-cc39-429d-8974-3dd4d64fc85f.mov

### CODE REVIEW

File: packages/frontend-shared/src/composables/useCollapsibleTree.ts [hunk 1]
```diff
@@ -43,6 +44,13 @@ export const useCollapsibleTreeNode = <T extends RawNode<T>>(rawNode: T, options
   const hidden = computed(() => {
     return !!roots.find((r) => r.expanded.value === false)
   })
+  const reveal = () => {
+    if (hidden.value) {
+      roots.forEach((root) => {
+        root.expanded.value = true
+      })
+    }
+  }
```
Comment (JessicaSachs) [inline packages/frontend-shared/src/composables/useCollapsibleTree.ts:53]:
```suggestion
```

File: packages/frontend-shared/src/composables/useCollapsibleTree.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ export type UseCollapsibleTreeNode <T extends RawNode<T>> = {
   hidden: ComputedRef<boolean>
   expanded: Ref<boolean>
   toggle: () => void
+  reveal: () => void
```
Comment (JessicaSachs) [inline packages/frontend-shared/src/composables/useCollapsibleTree.ts:14]:
```suggestion
```

File: packages/frontend-shared/src/composables/useCollapsibleTree.ts [hunk 3]
```diff
@@ -51,6 +59,7 @@ export const useCollapsibleTreeNode = <T extends RawNode<T>>(rawNode: T, options
     hidden,
     expanded,
     toggle,
+    reveal,
```
Comment (JessicaSachs) [inline packages/frontend-shared/src/composables/useCollapsibleTree.ts:62]:
```suggestion
```

**Review by lmiller1990 — DISMISSED** [review]

**Review by lmiller1990 — APPROVED** [review]
LGTM

Fixed tests for you

**Review by ZachJW34 — COMMENTED** [review]
So much better! One thing I noticed is that if your current spec hasn't been rendered (imagine choosing the very last spec and then you scroll to the top), you cannot tab into the list since no element has a tabIndex=0. Should it default to the first element in the list if the current spec is out of renderaable range?

**Comment by JessicaSachs** [comment]
> So much better! One thing I noticed is that if your current spec hasn't been rendered (imagine choosing the very last spec and then you scroll to the top), you cannot tab into the list since no element has a tabIndex=0. Should it default to the first element in the list if the current spec is out of renderable range?

I tried doing that behavior, but I couldn't get it to work and so I just took what I could 😢

**Pull request merged**